CONTRIBUTING.md

```diff
@@ -4,6 +4,8 @@ Thanks for your interest in improving Storybook! We are a community-driven proje
 
 Please review this document to help to streamline the process and save everyone's precious time.
 
+This guide assumes you're using `yarn` as package manager. You can use `npm` as well, but it would be less deterministic, as we only commit `yarn.lock` to the repo.
```

Comment:
We should be impartial and publish a `package-lock.json` as well imo. There's been problems with `npm` respecting their own lock files but that's been fixed in `npm@5.3`.

Reply:
If we're trying to making the installs more deterministic, I think we should be on the same page about which packager and script runner to use. Yarn and npm install trees are different and we shouldn't try to mix and match. For first time contributors setting up their dev environment, it could be quite a hassle.

There's also so many issues with the dependency trees in 5.0-5.2 that I'd realistically only suggest using >=5.3.

Reply:
Sorry, I can't get your point. If we make things better (by improving predictability) for a part of contributors (and mainly for CI), and it just doesn't affect others, why should we force those others into something?

Reply:
Our scripts call both `npm install` and `yarn install` right now and the user doesn't really have a choice as to using npm or yarn because it's predetermined by the bootstrap script. They are already forced to use `yarn` for everything and then `npm i` for the special react-native cases. I don't think they should be bootstrapping this project any other way.

https://github.com/storybooks/storybook/blob/master/package.json#L10

Reply:
Ok, the only difference then is in initial `install` in root package

Reply:
I think the only time someone will actually use the raw `npm` or `yarn` command is when doing `npm link` to link to an external project and `npm install --save` to add a new dependency. In the new dependency case, we should definitely use `yarn` since that will actually update the lock file.

I think for simplicity sake, we should just tell contributors to use `yarn`. I would love to support `npm` but the unnecessary complexity is really not worth it.

Reply:
> npm install --save

You can't do that with lerna

Reply:
> since that will actually update the lock file.

It will be updated after bootstrapping, which should be done anyway

Reply:
By the way, workspaces feature just became not experimental: https://github.com/yarnpkg/yarn/pull/4262

If we use it, there will be more sense in stating that we have a yarn-only setup. I'd prefer to introduce it in a separate PR, but I can adjust the wording here if we decide to go that way

Reply:
We develop storybook using yarn.

If someone wants to install the root with npm, that's of no consequence.. 
The bootstrap script will use yarn.

We'll assume everyone uses yarn to develop storybook in our documentation.